app/jinja_filters.py

```diff
@@ -43,6 +43,18 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_cond_date}} """
+
+    if dates[0]:
+        return format_str_as_short_date(dates[0])
+    else:
+        return format_date(dates[1])
```

Comment:
I think these should both use the same function otherwise the returned date will be formatted differently.
If the 2 values are in different formats then that just needs to be handled. 
This function shouldn't be implemented to just handle some data coming from metadata.